src/store/middleware/board.ts

```diff
@@ -1,127 +1,220 @@
+        });
+
+        subscription.on("open", (object) => {
+          joinRequestQuery.find().then((results) => {
+            dispatch(
+              ActionFactory.initializeJoinRequests(
+                (results as any[]).map(mapJoinRequestServerToClientModel)
+              )
+            );
+          });
+        });
+
+        subscription.on("create", (object) => {
+          dispatch(
+            ActionFactory.createJoinRequest(
+              mapJoinRequestServerToClientModel(
+                object as JoinRequestServerModel
+              )
+            )
+          );
+        });
+
+        subscription.on("update", (object) => {
+          dispatch(
+            ActionFactory.updateJoinRequest(
+              mapJoinRequestServerToClientModel(
+                object as JoinRequestServerModel
+              )
+            )
+          );
+        });
+      });
+    };
+
+    const createUsersSubscription = () => {
+      const adminsQuery = new Parse.Query(Parse.Role);
+      adminsQuery.equalTo("name", `admin_of_${action.boardId}`);
+      const memberQuery = new Parse.Query(Parse.Role);
+      memberQuery.equalTo("name", `member_of_${action.boardId}`);
+
+      const usersQuery = Parse.Query.or(adminsQuery, memberQuery);
+      usersQuery.subscribe().then((subscription) => {
+        closeSubscriptions.push(() => {
+          subscription.unsubscribe();
+        });
+
+        const updateUsers = (role: Parse.Role) => {
+          if (role.getName() === `member_of_${action.boardId}`) {
+            role
+              .getUsers()
+              .query()
+              .find()
+              .then((users) => {
+                dispatch(
+                  ActionFactory.setUsers(
+                    users.map((user) =>
+                      mapUserServerToClientModel(user.toJSON() as any, false)
+                    ),
+                    false
+                  )
+                );
+              });
+          }
+          if (role.getName() === `admin_of_${action.boardId}`) {
+            role
+              .getUsers()
+              .query()
+              .find()
+              .then((users) => {
+                dispatch(
+                  ActionFactory.setUsers(
+                    users.map((user) =>
+                      mapUserServerToClientModel(user.toJSON() as any, true)
+                    ),
+                    true
+                  )
+                );
+              });
+          }
+        };
+
+        subscription.on("update", (object) => {
+          updateUsers(object as Parse.Role);
+        });
+
+        subscription.on("open", () => {
+          usersQuery.find().then((results) => {
+            for (const result of results) {
+              updateUsers(result as Parse.Role);
+            }
+          });
+        });
+      });
+
+      const userQuery = new Parse.Query(Parse.User);
+      userQuery.equalTo("online", true);
+      userQuery.subscribe().then((subscription) => {
```

Comment:
I don't think that you need this query since the `update` of line 117 should also update by the user attributes? Not sure, but if not you should rather combine both queries.
Also if you combine the query the `setUserStatus` action becomes obsolete.

Reply:
I guess we do need them but maybe we can combine them. Let's do this together since I still need some help with Parse 😅

Reply:
finally done. Please check it again.